lib/kms/wrapper.js

```diff
@@ -0,0 +1,246 @@
+        return newKey;
+    }
+
+    /*
+     * cb(err, cipheredDataKey: Buffer)
+     */
+    static _cipherDataKey(cryptoScheme,
+                          masterKeyId,
+                          plainTextDataKey,
+                          log, cb) {
+        log.debug('ciphering a data key');
+        client.cipherDataKey(cryptoScheme, masterKeyId, plainTextDataKey, log,
+                             (err, cipheredDataKey) => {
+                                 if (err) {
+                                     log.warn('error from kms',
+                                              { implName, error: err });
+                                     return cb(err);
+                                 }
+                                 log.trace('data key ciphered by the kms');
+                                 return cb(null, cipheredDataKey);
+                             });
+    }
+
+    /*
+     * cb(err, plainTextDataKey: Buffer)
+     */
+    static _decipherDataKey(cryptoScheme, masterKeyId,
+                            cipheredDataKey, log, cb) {
+        log.debug('deciphering a data key');
+        client.decipherDataKey(cryptoScheme, masterKeyId, cipheredDataKey, log,
+                               (err, plainTextDataKey) => {
+                                   if (err) {
+                                       log.warn('error from kms',
+                                                { implName, error: err });
+                                       return cb(err);
+                                   }
+                                   log.trace('data key deciphered by the kms');
+                                   return cb(null, plainTextDataKey);
+                               });
+    }
+
+    /*
+     * cb(err, cipher: ReadWritable.stream)
+     */
+    static _createCipher(cryptoScheme, dataKey, offset, log, cb) {
+        log.debug('creating a cipher');
+        common.createCipher(
+            cryptoScheme, dataKey, offset, log,
+            (err, cipher) => {
+                if (err) {
+                    log.warn('error from kms', { implName, error: err });
+                    return cb(err);
+                }
+                log.trace('cipher created by the kms');
+                return cb(null, cipher);
+            });
+    }
+
+    /*
+     * cb(err, decipher: ReadWritable.stream)
+     */
+    static _createDecipher(cryptoScheme, dataKey, offset, log, cb) {
+        log.debug('creating a decipher');
+        common.createDecipher(
+            cryptoScheme, dataKey, offset, log,
+            (err, decipher) => {
+                if (err) {
+                    log.warn('error from kms', { implName, error: err });
+                    return cb(err);
+                }
+                log.trace('decipher created by the kms');
+                return cb(null, decipher);
+            });
+    }
+
+    /*
+     * cb(err, cipherBundle)
+     */
+    static createCipherBundle(serverSideEncryption,
```

Comment:
I think it would be clearer to call this serverSideEncryptionInfo as you did above.